pyradiosky/skymodel.py

```diff
@@ -1007,46 +1706,86 @@ def read_idl_catalog(filename_sav, expand_extended=True):
                 spectral_index = np.insert(spectral_index, use_index, src["alpha"])
                 source_inds = np.insert(source_inds, use_index, np.full(Ncomps, ext))
 
-    ra = Angle(ra, units.deg)
-    dec = Angle(dec, units.deg)
+    ra = Longitude(ra, units.deg)
+    dec = Latitude(dec, units.deg)
     stokes = stokes[:, np.newaxis, :]  # Add frequency axis
     sourcelist = SkyModel(
         ids,
         ra,
         dec,
         stokes,
-        source_freqs,
         spectral_type="spectral_index",
+        reference_frequency=Quantity(source_freqs, "hertz"),
         spectral_index=spectral_index,
         beam_amp=beam_amp,
         extended_model_group=extended_model_group,
     )
     return sourcelist
 
 
-def write_catalog_to_file(filename, catalog):
+def write_catalog_to_file(filename, skymodel):
```

Comment:
Certainly not now, but do we want to think about supporting other Stokes parameters in the text format? That would mean labeling columns as, say, "Flux_I_100MHz" and so on.

Reply:
I think we might in future, but I was trying to limit the number of changes. It would require deprecation even if we do it now..